tfjs-backend-webgpu/src/binary_op_util.ts

```diff
@@ -66,36 +66,14 @@ const GREATER_EQUAL = `
   let one = sign(b) * 0 + 1;
   let resultTemp = select(zero, one, a >= b);
 `;
-
 const INT_DIV = `
-  let s = sign(a) * sign(b);
-  let ia = i32(round(a));
-  let ib = i32(round(b));
-  return f32(idiv(ia, ib, s));
+  let numerator = round(a);
+  let denominator = round(b);
+  let remainder = round(numerator % denominator);
+  let quotient = round((numerator - remainder) / denominator);
+  let resultTemp =
+      select(quotient, quotient - 1, sign(remainder) == -sign(denominator));
```

Comment:
Several questions and comments:
1) round() in WGSL has a special definition: "Result is the integer k nearest to e, as a floating point value. When e lies halfway between integers k and k + 1, the result is k when k is even, and k + 1 when k is odd." Will this introduce a diff between CPU and WebGPU solution?
2) For "let remainder = round(numerator % denominator);" Is round() necessary?
3) What's the result of "9 / (-2)" in this code?
4) numerator -> dividend, denominator -> divisor?
5) Please ensure we have enough unit test cases for them, especially when dividend and/or divisor are halfway between integers, the results from "(numerator - remainder) / denominator" are halfway between integers, etc.

Reply:
> Several questions and comments:
> 
> 1. round() in WGSL has a special definition: "Result is the integer k nearest to e, as a floating point value. When e lies halfway between integers k and k + 1, the result is k when k is even, and k + 1 when k is odd." Will this introduce a diff between CPU and WebGPU solution?

This is not clearly mentioned in the TensorFlow.js API doc. Even if the CPU rounding mode differs from the GPU backend, I cannot tell which one is more correct, or if either is fine.

> 2. For "let remainder = round(numerator % denominator);" Is round() necessary?

No, but I feel that's more aligned with neighboring lines.

> 3. What's the result of "9 / (-2)" in this code?

-5

> 4. numerator -> dividend, denominator -> divisor?

The API doc prefers numerator/denominator.

> 5. Please ensure we have enough unit test cases for them, especially when dividend and/or divisor are halfway between integers, the results from "(numerator - remainder) / denominator" are halfway between integers, etc.

Reply:
Different languages may have different results.
[python]
(-9) % 2 -> 1
9 % (-2)  -> -1

[JavaScript]
(-9) % 2 -> -1
9 % (-2)  -> 1

So your result will be at least different from the JS's result. 
With round(), we introduce another level of difference between CPU and GPU, at least for the quotient calculation.

Reply:
Please note `INT_DIV` is exclusively used by `floorDiv()`, and the doc states the result should be floored. This matches with the current CPU behavior (`Math.floor(a / b)`).

Reply:
JiaJie, if `INT_DIV` is only used for `floorDiv`, will it be better to rename `INT_DIV` to `FLOOR_DIV`?